main.go

```diff
@@ -302,3 +337,13 @@ func setEnvironVar(env *[]string, key, value string) {
 
 	*env = append(*env, key+"="+value)
 }
+
+func shouldRetry(resp *http.Response) bool {
```

Comment:
I've read that code enough time that we might want some kind of baselib in go :)